src/services/ExposureNotificationService/ExposureNotificationService.ts

```diff
@@ -168,15 +170,32 @@ export class ExposureNotificationService {
     await this.recordKeySubmission();
   }
 
-  private async *keysSinceLastFetch(lastFetchDate?: Date): AsyncGenerator<string> {
+  private async *keysSinceLastFetch(lastFetchDate?: Date): AsyncGenerator<string | null> {
     const runningDate = new Date();
 
-    const lastCheckPeriod = periodSinceEpoch(lastFetchDate || addDays(runningDate, -14));
-    let runningPeriod = periodSinceEpoch(runningDate);
+    const lastCheckPeriod = periodSinceEpoch(lastFetchDate || addDays(runningDate, -14), hoursPerPeriod);
+    let runningPeriod = periodSinceEpoch(runningDate, hoursPerPeriod) - hoursPerPeriod;
 
     while (runningPeriod > lastCheckPeriod) {
-      yield await this.backendInterface.retrieveDiagnosisKeys(runningPeriod);
-      runningPeriod -= 2;
+      const zipFile = await this.backendInterface.retrieveDiagnosisKeys(runningPeriod);
+      if (zipFile) {
+        const components = zipFile.split('/');
+        components.pop();
+        components.push('keys-export');
+        const targetDir = components.join('/');
+
+        try {
+          const unzipResult = await unzip(zipFile, targetDir);
+          yield unzipResult;
+        } catch (err) {
```

Comment:
We should revisit this error handling once we're confident in the end-to-end flow.

Reply:
The unzip fails with this error: "cannot open file" and I believe the reason is the downloadTaskWithURL in CovidShield.m downloads the file into a temp location and once that process is done, the file gets deleted from the temp folder so when the unzip is happening, there is no file to unzip!

Reply:
No, the reason is different, I think I know what’s wrong. I’ll adjust, thanks for double checking that!

Reply:
I updated PR

Reply:
I just tested this change and it seems like it fixes the unzip issue. I am not sure if you guys have managed to do an end-to-end test with these recent changes but when I tested it, the app crashes on line 162 inside ExposureNotification.m with the following error: 
```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSCFString URLByDeletingPathExtension]: unrecognized selector sent to instance 0x282d921c0'
*** First throw call stack:
(0x1a528c300 0x1a4fa0c1c 0x1a518aa90 0x1a5290a60 0x1a5292d60 0x1dd2cb0c8 0x1dd2cae1c 0x1082d1d10 0x1082d318c 0x1082e1d14 0x1a52076b0 0x1a52022c8 0x1a52018f4 0x1af618604 0x1a93d5358 0x104fd824c 0x1a507d2dc)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

Reply:
this appears to be issue with decoding ExposureConfiguration from the server. I pushed change where it uses hardcoded configuration. 
it would still return an error from EN framework, what that error is I'm trying to figure out with apple engineer as we speak. 

sorry it takes this many iterations, I'm testing it in other branch and the copying changes over here. It's a very error prone process

Reply:
You guys are doing a wonderful job! Probably one of the best volunteer groups ever, with a noble cause!